tests/test_api.py

```diff
@@ -265,30 +265,51 @@ def test_event_error_rate(truth, prediction, eer):
 ])
 def test_concordance_cc(truth, prediction):
 
+    def expected(truth, prediction):
+        if len(prediction) < 2:
+            ccc_expected = np.NaN
+        else:
+            denominator = (
+                prediction.std() ** 2
+                + truth.std() ** 2
+                + (prediction.mean() - truth.mean()) ** 2
+            )
+            if denominator == 0:
+                ccc_expected = np.NaN
+            else:
+                r = np.corrcoef(list(prediction), list(truth))[0][1]
+                ccc_expected = (
+                    2 * r * prediction.std() * truth.std()
+                    / denominator
+                )
+        return ccc_expected
+
     ccc = audmetric.concordance_cc(truth, prediction)
 
     prediction = np.array(list(prediction))
     truth = np.array(list(truth))
 
-    if len(prediction) < 2:
-        ccc_expected = np.NaN
-    else:
-        denominator = (
-            prediction.std() ** 2
-            + truth.std() ** 2
-            + (prediction.mean() - truth.mean()) ** 2
-        )
-        if denominator == 0:
-            ccc_expected = np.NaN
-        else:
-            r = np.corrcoef(list(prediction), list(truth))[0][1]
-            ccc_expected = 2 * r * prediction.std() * truth.std() / denominator
-
     np.testing.assert_almost_equal(
         ccc,
-        ccc_expected,
+        expected(truth, prediction),
     )
 
+    # Handle NaN in prediction
```

Comment:
What happens if all values of `truth | prediction` are `NaN`?

Reply:
I switched now to your suggested

```python
if ignore_nan:
    mask = ~(np.isnan(truth) + np.isnan(prediction))
    truth = truth[mask]
    prediction = prediction[mask]
```

to make sure this is automatically handled and the code gets easier.
I have updated the benchmark results at the top. Using `ignore_nan=True` doubles roughly the execution time.

If all values are `NaN` then it is reduced to 0 samples. For 0 samples or 1 sample it will always return `NaN` as result.